File: typescript/pages/LoginPage.ts
import { Page, Locator, test } from '@playwright/test';

export class LoginPage {
    readonly usernameField: Locator;
    readonly passwordField:Locator; 
    readonly loginButton: Locator;

    constructor(page: Page) {
        this.usernameField = page.getByRole('textbox', { name: 'Username' });
        this.passwordField = page.getByRole('textbox', { name: 'Password' });
        this.loginButton = page.getByRole('button', { name: 'Login' });
    }

    async login(username: string, password: string) {
        await test.step('Fill Username field', async () => {
            await this.usernameField.fill(username);
        });

        await test.step('Fill Password field', async () => {
            await this.passwordField.fill(password);
        });

        await test.step('Click Login button', async () => {
            await this.loginButton.click();
        });
    }
}

File: python/pages/login_page.py
from playwright.sync_api import Locator, Page
import allure


class LoginPage:

    def __init__(self, page: Page):
        self.page: Locator = page
        self.username: Locator = page.get_by_role("textbox", name="Username")
        self.password: Locator = page.get_by_role("textbox", name="Password")
        self.login_button: Locator = page.get_by_role("button", name="Login")

    def login(self, username: str, password: str):
        with allure.step(f"Login as {username}"):
            self.username.fill(username)
            self.password.fill(password)
            self.login_button.click()
